File: 0543-diameter-of-binary-tree/0543-diameter-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int result = 0;

    public int getDiameter(TreeNode node){
        if(node == null)
            return 0;
        int left = getDiameter(node.left);
        int right = getDiameter(node.right);
        result = Math.max(result , right + left  );
        return Math.max(left,right) + 1;





    }
    public int diameterOfBinaryTree(TreeNode root) {
        getDiameter(root);
        return result;
        
        
    }
}

File: 0543-diameter-of-binary-tree/0543-diameter-of-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
 int res=-1;
 int height(struct TreeNode* node){
    if(node==0)
        return 0;
    int l=height(node->left);
    int r=height(node->right);
    int m=(l>r?l:r);
    res=(res>l+r?res:l+r);
    return 1+m ;
    
}
int diameterOfBinaryTree(struct TreeNode* root) {
    res=-1;
    height(root);
    return res;



    
}
